### PR TITLE
feat: Semantic viewer results runner

### PR DESCRIPTION
**Pull request by rephus**
Closes: 

### Description:

[Screencast from 15-08-24 08:46:38.webm](https://github.com/user-attachments/assets/4a76c3f9-385b-43e0-b424-26c0e7a1a0dd)

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Review by magnew — COMMENTED** [review]
Nice work, this is a lot to get working. My main concern is the duplication of all of the logic/code. I know you share that concern 😄. I think tactically it could make sense to merge this if we want to show fast progress now, but I feel its important that we spend some time in the near future getting things more shared. 

There are also some renames we should do for clarity, either now (since we are adding these files now) or soon after if we don't want to hold this out too long. 

Suggested renames:
- SemanticeViewerResultsViewerFE -> SemanticViewerRunnerFE
- TableDataTransformer -> TableModel
- useTableDataTransformer -> useTableModel
- SqlRunnerChart -> ChartView

And and similar ones.

File: packages/frontend/src/features/semanticViewer/store/semanticViewerSlice.ts
Comment (almeidabbm) [inline packages/frontend/src/features/semanticViewer/store/semanticViewerSlice.ts:193]: We should either always use `SemanticLayer` types for our slices or we will need to convert `SemanticLayerFieldType` into `DimensionType` here.

Casting it as `Dimension` type will cause certain fields to not show up in the selectors, e.g. if you select a time dimension you cannot set it in the X field of the charts

IMO the "best" solution would be to use the types in `packages/common/src/types/semanticLayer.ts` so we don't have to do a lot of type transformations in the frontend, but it would also be the one that would require the most work atm

Comment (IrakliJani) [reply packages/frontend/src/features/semanticViewer/store/semanticViewerSlice.ts:193]: @almeidabbm I have an PR which has a nice type abstractions on how we manage field states. we can merge this as is and once I rebase you can review the PR.

File: packages/frontend/src/features/semanticViewer/duckDBQuery.ts
Comment (rephus) [reply packages/frontend/src/features/semanticViewer/duckDBQuery.ts:1]: I think this is still used for pivoting, I have been debugging this to fix a bug with columns

File: packages/frontend/src/features/semanticViewer/components/ResultsViewer.tsx
Comment (rephus) [reply packages/frontend/src/features/semanticViewer/components/ResultsViewer.tsx:115]: yeah, I think that would be better, it depends if we need in some other places the full list of columns or just the selected columns. but we can add 2 states in the slice .

File: packages/frontend/src/features/semanticViewer/store/barChartSlice.ts
Comment (rephus) [reply packages/frontend/src/features/semanticViewer/store/barChartSlice.ts:16]: the chart slices are almost identical, so we can probably share them.

File: packages/frontend/src/features/semanticViewer/store/semanticViewerSlice.ts
Comment (rephus) [reply packages/frontend/src/features/semanticViewer/store/semanticViewerSlice.ts:193]: yeah, I had to force this type to make it work with the sql runner transformer, we should use semanticLayerField or build a proper conversion between types.

**Review by almeidabbm — APPROVED** [review]

**Pull request merged**
